src/app/basketProduct/basketProduct.component.ts

```diff
@@ -25,9 +25,11 @@ export class BasketProductComponent implements OnInit {
   getProducts(): BasketProduct[] {
     return this.basketProductService.getAllProducts();
   }
+
   deleteProduct(product) {
     this.basketProductService.removeProduct(product);
     this.totalPrice = this.getTotalPrice();
+    window.location.reload();
```

Comment:
Avoid reloading the page at all costs when working with one-page applications.
In this case, the solution is simply to update the `this.products` after deleting an element. Angular will automatically update the value for you in the template(s) that are observing this value.